### DISCUSSION TITLE
Advice upgrading to 3.6

### DISCUSSION THREAD

**Discussion by akhial** [saleor/saleor#10608] [Deployment] [answered]
Hi Saleor team!

We are currently on Saleor 3.4 and thinking about upgrading to 3.6, we have a question regarding the new thumbnails behavior.
We currently have `CREATE_IMAGES_ON_DEMAND` set to False. We are hosting media on Google Cloud Storage.

How could we go about upgrading to 3.6, is it just enough to update our usage of the API and remove usages of `create_thumbnails`?
And can we safely delete the `__sized__` directory on GCS?

Also are we correct to assume `CREATE_IMAGES_ON_DEMAND` no longer does anything?

Thanks for the help.

**Comment by IKarbowiak** [comment, marked as answer]
Hi @akhial,

In 3.6 thumbnails are created when the picture is opened for the first time and it's saved on default storage, so in your case, it will be in  Google Cloud Storage. The next time user asks for the same thumbnail, the previously generated thumbnail is returned. You can read more about the new approach [here](https://docs.saleor.io/docs/3.x/developer/thumbnails#generating-the-thumbnails).

> How could we go about upgrading to 3.6, is it just enough to update our usage of the API and remove usages of create_thumbnails?
And can we safely delete the __sized__ directory on GCS?

Yes, the steps that you describe should be enough. The picture will be stored in the `thumbnails` directory so the `__sized__` could be deleted. And as you correctly noticed `CREATE_IMAGES_ON_DEMAND` no longer does anything as all thumbnails are created on demand.
